app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    @user = User.from_omniauth(auth)
+
+    if @user.persisted?
+      flash[:notice] = I18n.t("devise.omniauth_callbacks.success",
+                              kind: "Google")
+      sign_in_and_redirect @user, event: :authentication
+    else
+      session["devise.google_data"] = request.env["omniauth.auth"]
+                                             .except("extra")
+      redirect_to new_user_registration_url, alert: @user.errors.full_messages
+                                                         .join("\n")
+    end
+  end
+
+  private
+
+  def auth
```

Comment:
Anh thấy cái này chỉ dùng ở 1 chỗ phía trên nên gọi thẳng tên luôn cx đc. Ko cần đặt hàm đâu

Reply:
Assignment Branch Condition size for google_oauth2 is too high
xóa đi bị lỗi này anh ạ

Reply:
Cái này là lỗi rubucop đúng ko